src/java.base/share/classes/java/lang/ModuleLayer.java

```diff
@@ -297,6 +299,35 @@ public Controller addOpens(Module source, String pn, Module target) {
             source.implAddOpens(pn, target);
             return this;
         }
+
+        /***
+         * Enables native access for a module in the layer if the caller's module
+         * already has native access.
```

Comment:
The API looks okay, ModuleLayer.Controller is the right place for this. I think you can drop "already" from the javadoc.